File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 1) return 0;
        
        int out = 0;
        int max = prices[0];
        int min = prices[0];

        for (int p : prices) {
            if (p - min > out) {
                max = p;
                out = max - min;
            } else if (p < min) {
                min = p;
            }
        }

        return out;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights)-1
        max_area = (r-l) * min(heights[l],heights[r])
        
        while l<r:
            if heights[l]<heights[r]:
                l += 1
            elif heights[r]<heights[l]:
                r -= 1
            else:
                l += 1
                r -= 1
            area = (r-l) * min(heights[l],heights[r])
            max_area = max(max_area, area)

        return max_area

             